src/bp/nlu-server/validation/validate.test.ts

```diff
@@ -0,0 +1,373 @@
+import { Enum, Intent, Pattern, Topic, TrainInput, Variable } from '../typings'
+
+import validateInput from './validate'
+
+/**
+ * These unit tests don't cover all possible scenarios of training input, but they to more good than bad.
+ * If we ever find a bug in train input validation, we'll just add some more tests.
+ */
+
+async function assertThrows(fn: () => Promise<any>, msg: string = '') {
```

Comment:
that's builtin in jest testing framework

Reply:
and if you're wondering how it works with async code, check this out : https://jestjs.io/docs/en/asynchronous#resolves-rejects

Reply:
for some reason, I'm not sure why but it kept throwing... I'll double check, its been a while since I've used it.

Reply:
Ok you're right, it works well!